File: .claude/skills/zero-trust-secret-orchestration/references/mjs/src/strategy.mjs
export class Strategy {
  /**
   * Resolve a secret value by key. Concrete strategies override this.
   *
   * Return contract:
   *   - `null`   : "I don't have this key" — resolver continues the chain.
   *   - throw    : "I had it but resolution failed" — resolver STOPS here.
   *
   * @param {string} key
   * @returns {Promise<string | null>}
   */
  async get(key) {
    throw new Error(`${this.constructor.name}.get(key) is not implemented`);
  }
}

export class SecretResolutionError extends Error {
  constructor(message, { cause } = {}) {
    super(message);
    this.name = "SecretResolutionError";
    if (cause !== undefined) this.cause = cause;
  }
}

export class StrategyConfigError extends Error {
  constructor(message) {
    super(message);
    this.name = "StrategyConfigError";
  }
}

File: .claude/skills/zero-trust-secret-orchestration/references/py/src/zts/strategy.py
from abc import ABC, abstractmethod


class Strategy(ABC):
    """Polymorphic base for secret resolution strategies.

    Return contract for `get(key)`:
        - return None: I don't have this key; resolver continues the chain.
        - raise:       I had it but resolution failed; resolver STOPS here.
    """

    @abstractmethod
    async def get(self, key: str) -> str | None:
        raise NotImplementedError


class SecretResolutionError(Exception):
    """Raised when a strategy had the key but resolution failed.

    Resolver chain stops here; caller decides whether to retry.
    """


class StrategyConfigError(Exception):
    """Raised at resolver construction when the chain is mis-configured."""
